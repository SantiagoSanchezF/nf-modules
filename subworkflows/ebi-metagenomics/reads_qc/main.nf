
include { SEQFU_CHECK } from '../../../modules/ebi-metagenomics/seqfu/check/main'
include { FASTP      } from '../../../modules/ebi-metagenomics/fastp/main'
include { SEQTK_SEQ     } from '../../../modules/ebi-metagenomics/seqtk/seq/main'

workflow  READS_QC {

    take:
    ch_reads    // channel: [ val(meta), [ fastq ] ]
    save_merged // channel:  val(boolean)

    main:
    ch_versions = Channel.empty()
    
    FASTP ( ch_reads, params.save_trimmed_fail, save_merged )
    ch_versions = ch_versions.mix(FASTP.out.versions.first())

    ch_se_fastp_reads = FASTP
                        .out.reads
                        .filter { it[0].single_end }

    ch_reads_se_and_merged = ch_se_fastp_reads
                            .mix(FASTP.out.reads_merged)

    SEQTK_SEQ(ch_reads_se_and_merged)
    ch_versions = ch_versions.mix(SEQTK_SEQ.out.versions.first())

    emit:
    reads               = FASTP.out.reads           // channel: [ val(meta), [ fastq ] ]
    reads_se_and_merged = ch_reads_se_and_merged    // channel: [ val(meta), [ fastq ] ]
    fastp_summary_json  = FASTP.out.json            // channel: [ val(meta), [ json ] ]
    reads_fasta         = SEQTK_SEQ.out.fastx       // channel: [ val(meta), [ fasta ] ]
    versions            = ch_versions               // channel: [ versions.yml ]
}

