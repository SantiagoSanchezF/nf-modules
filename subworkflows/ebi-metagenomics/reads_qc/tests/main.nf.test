nextflow_workflow {

    name "Test Subworkflow READS_QC"
    script "../main.nf"
    workflow "READS_QC"
    config "./nextflow.config"

    tag "subworkflows"
    tag "subworkflows_nfcore"
    tag "subworkflows/reads_qc"
    tag "fastp"
    tag "seqtk"
    tag "seqtk/seq"

    test("test_reads_qc") {

        when {
            params {
                save_trimmed_fail = true
            }
            workflow {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    [ file("${moduleDir}/tests/fixtures/SRR21814853_1.fastq.gz", checkIfExists: true),
                      file("${moduleDir}/tests/fixtures/SRR21814853_2.fastq.gz", checkIfExists: true)
                    ]
                    ]
                input[1] = true
                """
            }
        }
        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }            
            )
        }
    }
}
