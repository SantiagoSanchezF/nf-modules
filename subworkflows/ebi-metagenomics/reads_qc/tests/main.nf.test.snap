{
    "test_reads_qc": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test_1.fastp.fastq.gz:md5,ab81a64d069ed8876432d8e3bad0b847",
                            "test_2.fastp.fastq.gz:md5,f8bf866d116205baa454b7a501a67b62"
                        ]
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.merged.fastq.gz:md5,58ededca56f023aa625f51ec241ba849"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.fastp.json:md5,552903446c04b38bd8f2f8976119fc42"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.seqtk-seq.fasta.gz:md5,a14b76b1e8c24d3f9d98e61d03a62a53"
                    ]
                ],
                "4": [
                    "versions.yml:md5,a5fd01ddb43380f5a73a5fcdb4dde637",
                    "versions.yml:md5,caf2b2e3fac12c4d49e72d81275628ce"
                ],
                "fastp_summary_json": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.fastp.json:md5,552903446c04b38bd8f2f8976119fc42"
                    ]
                ],
                "reads": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test_1.fastp.fastq.gz:md5,ab81a64d069ed8876432d8e3bad0b847",
                            "test_2.fastp.fastq.gz:md5,f8bf866d116205baa454b7a501a67b62"
                        ]
                    ]
                ],
                "reads_fasta": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.seqtk-seq.fasta.gz:md5,a14b76b1e8c24d3f9d98e61d03a62a53"
                    ]
                ],
                "reads_se_and_merged": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.merged.fastq.gz:md5,58ededca56f023aa625f51ec241ba849"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,a5fd01ddb43380f5a73a5fcdb4dde637",
                    "versions.yml:md5,caf2b2e3fac12c4d49e72d81275628ce"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-02T11:34:31.175590748"
    }
}